tests/conftest.py

```diff
@@ -0,0 +1,76 @@
+import os
+import re
+import subprocess
+import sys
+from pathlib import Path
+
+import pytest
+
+
+TESTS_PATH = Path(__file__).parent
+IMAGE = "stata-mp"
+
+
+RE_WHITESPACE = re.compile(r"\s+")
+
+
+def sanitize(string):
+    """
+    Replace all whitespaces in a string with single whitespace, to make asserting
+    content more robust
+    """
+    return RE_WHITESPACE.sub(" ", string).strip()
+
+
+@pytest.fixture(autouse=True)
+def cleanup():
+    """Remove any old log or output files"""
+    yield
+    log_files = TESTS_PATH.glob("*.log")
+    output_files = TESTS_PATH / "output"
+    for log_file in log_files:
+        log_file.unlink()
+    for output_file in output_files.iterdir():
+        if output_file.name != ".keep":
+            output_file.unlink()
+
+
+@pytest.fixture(autouse=True)
+def check_stata_license():
+    if "STATA_LICENSE" not in os.environ:
+        print("No STATA_LICENSE detected")
+        sys.exit(1)
+
+
+@pytest.fixture
+def run_docker():
+    def _run(command):
+        filestem = Path(command.split()[0]).stem
+        user = subprocess.check_output(
+            "echo $(id -u):$(id -g)", shell=True, universal_newlines=True
+        ).strip("\n")
+        process = subprocess.run(
+            f"docker run --rm --user {user} -e STATA_LICENSE -v {TESTS_PATH.resolve()}:/workspace "
+            f"{IMAGE} {command}",
+            shell=True,
```

Comment:
nit: I prefer cmd lists and not setting shell=True (which can often trip linting errors), but this is a test so its probably fine, and just my OCD :)

Reply:
I tried to do this and couldn't get it to work, but I've just realised why (the `command` in some tests is a string with spaces and needed split)